compiler/rustc_const_eval/src/interpret/operator.rs

```diff
@@ -308,6 +308,13 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             BitAnd => ImmTy::from_uint(l & r, left.layout),
             BitXor => ImmTy::from_uint(l ^ r, left.layout),
 
+            BitOrDisjoint => {
+                if l & r != 0 {
+                    todo!(); // Can't figure out what to do here based on the surrounding code
```

Comment:
You need to `throw_ub!`. To that end, you first need to add a new error variant to `UndefinedBehaviorInfo`, and then use that here.

Make sure to also add a Miri testcase that checks that we detect the UB here.